clients/src/test/java/org/apache/kafka/common/network/SslTransportLayerTest.java

```diff
@@ -580,7 +581,16 @@ public void testTLSDefaults() throws Exception {
 
     @Test
     public void testUnsupportedCipher() throws Exception {
-        String[] cipherSuites = ((SSLServerSocketFactory) SSLServerSocketFactory.getDefault()).getSupportedCipherSuites();
+        String[] cipherSuites;
+        if (Java.IS_JAVA11_COMPATIBLE) {
+            cipherSuites = new String[] {
+                "TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384",
```

Comment:
We should use correct cipher for the server(which uses `TLSv1.3` in case Java11) otherwise server metrics not updated because we never get `AuthenticationException` on the server-side. Instead of it, we get `IOException` from `SSLTransportLayer`:

```
    private SSLEngineResult handshakeUnwrap(boolean doRead, boolean ignoreHandshakeStatus) throws IOException {
...
        // Throw EOF exception for failed read after processing already received data
        // so that handshake failures are reported correctly
        if (read == -1)
            throw new EOFException("EOF during handshake, handshake status is " + handshakeStatus);
}
```

Reply:
The previous logic is weird, I agree. I think the idea here is to simply pick a different supported cipher in the server vs the client. I think we can drop `getSupportedCipherSuites` altogether and just pick two ciphers explicitly. The cipher names would be different for TLS 1.2 versus TLS 1.3.

Also, we should change the following to simply use the `tlsVersion` field.

```
            checkAuthentiationFailed("1", "TLSv1.1");
            server.verifyAuthenticationMetrics(0, 1);
            checkAuthentiationFailed("2", "TLSv1");
            server.verifyAuthenticationMetrics(0, 2);
```

Reply:
Also, note that we also have `testUnsupportedCiphers`. I wonder if we can delete this test.

Reply:
Maybe we add `checkAuthentiationFailed` to `testUnsupportedCiphers` and delete this test.

Reply:
Done.